### DISCUSSION TITLE
Announce R package 'cgalMeshes'

### DISCUSSION THREAD

**Discussion by stla** [CGAL/cgal#7002] [Show and tell]
For R users: ['CGAL' meets 'R6': the 'cgalMeshes' package](https://laustep.github.io/stlahblog/posts/cgalMeshes.html).

**Comment by lrineau** [comment]
Great! Congratulation for that work you've done and that milestone you achieved. I hope the package will soon get accepted into the CRAN archive. 

I know you have been struggling with CGAL code in the past, and I saw that paragraph on you blog post:
> I already wrapped some CGAL stuff in some other packages. But these packages have been archived by CRAN because there were some warnings from the CGAL library, and I can’t fix them. So I didn’t implement the algorithms which throw some CRAN-unacceptable warnings in cgalMeshes, and hopefully this package will remain on CRAN.

Maybe we should work together, you and the CGAL team, to try and eventually solve those unacceptable warnings, so that you can progressively re-add other functionality to your `cgalMeshes` package.

**Reply by stla** [reply]
Hello @lrineau 

Thanks for your reply. CRAN is very strict. They check the packages with some tools such as Valgrind and they don't accept any warning from these tools.

Here are some issues:

- [segfault with the Minkowski sum](https://github.com/CGAL/cgal/discussions/6994) (just a simple cube + a simple octahedron)

- [warning with the Poisson reconstruction](https://github.com/CGAL/cgal/discussions/6793)

**Comment by stla** [comment]
Another package in progress: [lazyNumbers](https://github.com/stla/lazyNumbers).

**Comment by stla** [comment]
Update: <https://laustep.github.io/stlahblog/posts/cgalMeshes2.html>.

**Comment by stla** [comment]
Now [cgalPolygons](https://github.com/stla/cgalPolygons). 

CRAN currently has an issue with CGAL, related to **mpfr**. The compilation fails on Windows. Anyway I'll do some Github-only versions of my packages.

**Reply by lrineau** [reply]
With CGAL-5.6 (current `master`), GMP and MPFR will become optional (replaced by the header-only Boost.Mulprecision library). That might solve your issues.

**Reply by stla** [reply]
Cool. It remains to convince the author of 'RcppCGAL' to upgrade.

**Reply by lrineau** [reply]
Note that CGAL-5.6 is not yet published. It is planned for summer 2023. The `master` is tested and ready for production, but that is not an official release with a number (it's official name is `CGAL-5.6-dev`).

**Reply by stla** [reply]
Ok. But does that mean that **boost::multiprecision** does not longer depend on **gmp**? Because it did.
